### ISSUE TITLE
Invalid typed array lenght

### ISSUE THREAD

**Issue by rumo19** [justinwilaby/sax-wasm#9]
When I works with small size of xml (<500kb), it works fine. But with big size of xml, it showed me this error. 

Uncaught RangeError: Invalid typed array length: 2428901
    at new Uint8Array (<anonymous>)
    at SAXParser.write (saxWasm.js:63)
    at XMLHttpRequest.xhttp.onreadystatechange (index.js:49)
How could i deal with this problem?
Thank you.

**Comment by justinwilaby** [comment]
Thank you for filing this issue! It reminds me that I need to provide docs for parsing very large files. 

The best way to work around this is to break up the file into smaller chunks and write those chunks to the parser. For instance:

```js
const largeXml = fs.readFileSync('some/path/to/large.xml', 'utf8');
const len = largeXml.length;
const chunkSize = 1000;
let idx = 0;
while (idx < len) {
  parser.write(largeXml.substr(idx, chunkSize);
  idx += chunkSize;
}
parser.end();
```
Something like this should work better than writing the entire file at once. Let me know if this works!

**Comment by rumo19** [comment]
Hallo Justin,
Thank you.
Up-to to a specific point , It parsed the xml file. But when it got this error, it did not parse further. 
**Uncaught SyntaxError: Unexpected token > in JSON at position 84**
    at JSON.parse (<anonymous>)
    at SAXParser.eventTrap (saxWasm.js:42)
    at core::ops::function::FnOnce::call_once::hdb5338efa9368c36 (:8081/wasm-function[52]:7)
    at main::sax::parser::SAXParser::process_open_tag::h639325b6152a21ae (:8081/wasm-function[57]:1779)
    at main::sax::parser::SAXParser::process_grapheme::hdaacd3dc25e8c041 (:8081/wasm-function[54]:6254)
    at write (:8081/wasm-function[53]:106)
    at SAXParser.write (saxWasm.js:65)

**Comment by justinwilaby** [comment]
Can you set a breakpoint [here](https://github.com/justinwilaby/sax-wasm/blob/master/src/js/saxWasm.ts#L101) and post the full string? It looks like there could be an unescaped char that the JSON parser doesn't like. It should be a fairly easy fix.

**Comment by rumo19** [comment]
This is the last result before doing any further parsing
{"name":"ShipInfo","attributes":[{"name":"ShippedDate","**value":"1997-06-26T00:00:00">**  ","nameStart":{"line":1096,"character":16},"nameEnd":{"line":1096,"character":27},"valueStart":{"line":1096,"character":29},"valueEnd":{"line":1097,"character":4294967295}},{"name":"ShipVia","value":"","nameStart":{"line":1097,"character":9},"nameEnd":{"line":0,"character":0},"valueStart":{"line":0,"character":0},"valueEnd":{"line":0,"character":0}}],"openStart":{"line":1096,"character":6},"openEnd":{"line":1097,"character":17},"closeStart":{"line":0,"character":0},"closeEnd":{"line":0,"character":0},"textNodes":[],"selfClosing":false}

**Comment by rumo19** [comment]
When I delete this line from xml, it works fine

**Comment by rumo19** [comment]
What changes should  be made  in this variable rawUtf8String ? How could you fix it? Should I use replace function to skip this special char.

**Comment by justinwilaby** [comment]
You can use `rawUtf8String.replace('"', '\\"');` until I get a fix completed.

**Comment by rumo19** [comment]
ok, Thanks.

**Comment by rumo19** [comment]
Another things:
I also did same parsing with sax-js. It seems like sax-js is faster than sax-wasm. I used 100KB, 20KB files for testing. I used performance.now() to track the timings.Is there any way I could check the performance of sax-wasm. 
 Could you tell me somethings about it?

**Comment by rumo19** [comment]
Thank you :+1: 
I just run 86MB files. It works fine. For the error, I think it is the formation of xml. For some format of xml , it works and for others, It does not. 
I will give a presentation about xml parsing with wasm in my university. I will take your project as primary resources.  If you tell me how wasm works in your project and also interact with the web browser, It will be helpful for my presentation. Your project is very interesting. Hopefully, try to work in future

**Comment by justinwilaby** [comment]
I would be honored! Please let me know how I can help. Also, I have a fix in [pr #10](https://github.com/justinwilaby/sax-wasm/pull/10)
